src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/services/DomainEventSubscriber.kt

```diff
@@ -31,7 +31,7 @@ class DomainEventSubscriber(
       when (eventType) {
         "prison-offender-events.prisoner.merged" ->
           restrictedPatientCleanup.mergeRestrictedPatient(
-            additionalInformation.removedNomsNumber,
+            additionalInformation.removedNomsNumber!!,
```

Comment:
for a merge the removed will always be populated - problem is here that `additionalInformation` shared between different message types